vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java

```diff
@@ -175,6 +174,9 @@ public void setValueChangeMode(ValueChangeMode valueChangeMode) {
         }
 
         super.setValueChangeMode(valueChangeMode);
+
+        getSynchronizationRegistration()
+                .synchronizeProperty("_inputElementValue");
```

Comment:
Spent some time trying to understand everything works together. What I got so far is that this should synchronize `_inputElementValue` whenever `value` changes. However that wouldn't cover cases where `value` doesn't change, for example when having an empty input and then entering an unparsable value. This can be fixed by synchronizing the property for the `unparsable-change` event listener as well. Also looks like the unit tests will not be sufficient to cover this.

Reply:
Thanks, it turns out it was caught by existing ITs as well. Fixed by synchronizing the property on `unparsable-change`.